pronto_ros/include/pronto_ros/ros_frontend.hpp

```diff
@@ -410,10 +412,20 @@ std::cerr << ":::::::" << std::endl;
             // fill in time
             pose_msg_.header.stamp = ros::Time().fromNSec(head_state.utime * 1000);
             if(publish_tf_){
-                tf_pose_.setOrigin(temp_v3);
-                tf_pose_.setRotation(temp_q);
-                tf_pose_.stamp_ = ros::Time::now();
-                tf_broadcaster_.sendTransform(tf_pose_);
+                // Only publish the pose if the timestamp is different:
+                // This prevents issues in Noetic where repeated warnings of type:
+                // "TF_REPEATED_DATA ignoring data with redundant timestamp for frame base at time"
+                // are otherwise printed to the terminal.
+                // Cf. https://github.com/ros/geometry2/issues/467#issuecomment-751572836
+                ros::Time new_stamp = ros::Time::now();
+                if (new_stamp > transform_msg_.header.stamp) {
+                    transform_msg_.transform.translation.x = pose_msg_.pose.pose.position.x;
+                    transform_msg_.transform.translation.y = pose_msg_.pose.pose.position.y;
+                    transform_msg_.transform.translation.z = pose_msg_.pose.pose.position.z;
+                    transform_msg_.transform.rotation = pose_msg_.pose.pose.orientation;
+                    transform_msg_.header.stamp = new_stamp;
```

Comment:
@mcamurri I wasn't sure if the stamp of the transform shouldn't be the same as in the `pose_msg_` (from `head_state`):

```cpp
pose_msg_.header.stamp = ros::Time().fromNSec(head_state.utime * 1000);
```

I preserved how it was previously done but wonder which one is correct?